app/src/main/java/org/sil/storyproducer/controller/SelectTemplatesFolderController.kt

```diff
@@ -18,32 +20,61 @@ class SelectTemplatesFolderController(
         view.startActivityForResult(intent, request)
     }
 
+    private fun showMigrateIntenalStoriesDialog(request: Int, uri: Uri) {
+        val dialog = AlertDialog.Builder(context)
+            .setTitle(context.getString(R.string.migrate_stories_title))
+            .setMessage(context.getString(R.string.migrate_stories_message))
+            .setPositiveButton(context.getString(R.string.yes)) { _, _ ->
+                setupWorkspace(request, uri, true)
+            }
+            .setNegativeButton(context.getString(R.string.no)) { _, _ ->
+                setupWorkspace(request, uri, false)
+            }
+            .setOnCancelListener() {
+                setupWorkspace(request, uri, false)
+            }
+            .create()
+
+        dialog.show()
+    }
+
     fun onFolderSelected(request: Int, result: Int, data: Intent?) {
         data?.data?.also { uri ->
             if (result == Activity.RESULT_OK) {
-                setupWorkspace(request, uri)
+                if (!workspace.setupWorkspacePath(context, uri, false))
+                    return
+                if (//workspace.workdocfile.uri.scheme == "file" && // TODO: Should we only migrate from internal folders?
+                        workspace.previousWorkDocFile.exists() &&
+                        workspace.previousWorkDocFile.uri != workspace.workdocfile.uri &&
+                        workspace.oldStoryDirectories(workspace.storyDirectories(), true).isNotEmpty()) {
```

Comment:
checks that there exist stories in the old folder that do not exist in the new selected folder

Reply:
A comment has been added to 'Automatic workspace folder' document to note that this is still TBD but migrating from any previous folder is enabled for now.